File: solutions/problem-17/solution.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>
void num(long long a)
{
    int p,q,r;
    p=a/100;
    if(p>0)
        {
        switch(p)
            {
    case 1:
        printf("One ");
        break;
    case 2:
        printf("Two ");
        break;
    case 3:
        printf("Three ");
        break;
    case 4:
        printf("Four ");
        break;
    case 5:
        printf("Five ");
        break;
    case 6:
        printf("Six ");
        break;
    case 7:
        printf("Seven ");
        break;
    case 8:
        printf("Eight ");
        break;
    case 9:
        printf("Nine ");
        }
        printf("Hundred ");
    }
      p=a%100;
    if(p<=20)
        {
        switch(p)
            {
    case 1:
        printf("One ");
        break;
    case 2:
        printf("Two ");
        break;
    case 3:
        printf("Three ");
        break;
    case 4:
        printf("Four ");
        break;
    case 5:
        printf("Five ");
        break;
    case 6:
        printf("Six ");
        break;
    case 7:
        printf("Seven ");
        break;
    case 8:
        printf("Eight ");
        break;
    case 9:
        printf("Nine ");
    case 10:
        printf("Ten ");
        break;
    case 11:
        printf("Eleven ");
        break;
    case 12:
        printf("Twelve ");
        break;
    case 13:
        printf("Thirteen ");
        break;
    case 14:
        printf("Fourteen ");
        break;
            case 15:
        printf("Fifteen ");
        break;
            case 16:
        printf("Sixteen ");
        break;
            case 17:
        printf("Seventeen ");
        break;
            case 18:
        printf("Eighteen ");
        break;
            case 19:
        printf("Nineteen ");
        break;
            case 20:
        printf("Twenty ");
        break;
        }
    }
    if(p>20)
        {
        q=p/10;
        r=p%10;
        switch(q)
            {
    case 2:
        printf("Twenty ");
        break;
    case 3:
        printf("Thirty ");
        break;
    case 4:
        printf("Forty ");
        break;
    case 5:
        printf("Fifty ");
        break;
    case 6:
        printf("Sixty ");
        break;
    case 7:
        printf("Seventy ");
        break;
    case 8:
        printf("Eighty ");
        break;
    case 9:
        printf("Ninety ");
        }
        switch(r)
       {
    case 1:
        printf("One ");
        break;
    case 2:
        printf("Two ");
        break;
    case 3:
        printf("Three ");
        break;
    case 4:
        printf("Four ");
        break;
    case 5:
        printf("Five ");
        break;
    case 6:
        printf("Six ");
        break;
    case 7:
        printf("Seven ");
        break;
    case 8:
        printf("Eight ");
        break;
    case 9:
        printf("Nine ");
        }
        
    }
}
int main() 
{
    int t,i,j;
    long long n;
    scanf("%d",&t);
    for(i=0;i<t;i++)
        {
        scanf("%Ld",&n);
        if(n==0)
            printf("Zero\n");
        else
       {
        if(n>=pow(10,9))
         {
            num(n/(long long)pow(10,9));
            printf("Billion ");
        }
        n=n%(long long)pow(10,9);
        if(n>=pow(10,6))
         {
            num(n/(long long)pow(10,6));
            printf("Million ");
        }
        n=n%(long long)pow(10,6);
        if(n>=pow(10,3))
         {
            num(n/(long long)pow(10,3));
            printf("Thousand ");
        }
        n=n%(long long)pow(10,3);
        num(n);
        printf("\n");
    }
    }    
    return 0;
}

File: solutions/problem-17/solution.py
#solution to problem 17
from random import randint

digit_to_word = {
"1": "one",
"2": "two",
"3": "three",
"4": "four",
"5": "five",
"6": "six",
"7": "seven",
"8": "eight",
"9": "nine"
}

thy_to_word = {
"2": "twenty",
"3": "thirty",
"4": "forty",
"5": "fifty",
"6": "sixty",
"7": "seventy",
"8": "eighty",
"9": "ninety"
}

teen_to_word = {
"10": "ten",
"11": "eleven",
"12": "twelve",
"13": "thirteen",
"14": "fourteen",
"15": "fifteen",
"16": "sixteen",
"17": "seventeen",
"18": "eighteen",
"19": "nineteen"
}

hundred = "hundred"
thousand = "thousand"

def number_to_english(x):
	if(len(str(x))>=5):
		return "Too big number"

	thousand_str = None
	hundred_str = None
	ten_str = None
	digit_str = None

	if(len(str(x))>=4) and (str(x)[-4]!="0"):
		thousand_str = digit_to_word[str(x)[-4]] + " " + thousand
	
	if(len(str(x))>=3) and (str(x)[-3]!="0"):
		hundred_str = digit_to_word[str(x)[-3]] + " " + hundred

	if( ((x-round(x/100)*100)>=10) and ((x-round(x/100)*100)<20)):
		last_two = str(x)[-2:]
		digit_str = teen_to_word[last_two]
	else:
		if(len(str(x))>=2) and (str(x)[-2]!="0"):
			ten_str = thy_to_word[str(x)[-2]]

		if(len(str(x))>0) and (str(x)[-1]!="0"):
			digit_str = digit_to_word[str(x)[-1]]

	res = ""
	if(thousand_str):
		res = thousand_str + " "
	if(hundred_str):
		res = res + hundred_str + " "
	if (thousand_str or hundred_str) and (ten_str or digit_str):
		res = res + "and "
	if ten_str:
		res = res + ten_str
	if ten_str and digit_str:
		res = res + "-"
	if digit_str:
		res = res + digit_str

	return res

sum = 0

for x in xrange(1,1001):
	sum = sum + len(number_to_english(x).replace('-', '').replace(' ', ''))

print(sum)
